### ISSUE TITLE
Cannot run provider verification against HTTPS endpoint that has CA, if called from a http endpoint

### ISSUE THREAD

**Issue by YOU54F** [pact-foundation/pact-js#281]
For the example provided in https://github.com/pact-foundation/pact-js/issues/280, we were attempting to hit providerBaseUrl with a HTTPS end-point, self-signed without a CA. This is now working with the v8.0.4 release and the introduction of `validateSSL` flag.

In our work, we were attempting to hit providerBaseUrl with a HTTPS end-point, self-signed but with a CA, so we need `validateSSL` flag to set to true. I was unable to hit it and was getting the following error 

```
Error: write EPROTO 4515927488:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1407:SSL alert number 40
```

I had a google and found a few similar issues but this thread had the answer that has resolved it

https://stackoverflow.com/questions/14262986/node-js-hostname-ip-doesnt-match-certificates-altnames

specifically this

```
n case you are using http-proxy package this happens when your server is HTTP (for example localhost) and target is HTTPS. To fix this issue set changeOrigin to true.

const proxy = httpProxy.createProxyServer();

proxy.web(req, res, {
  changeOrigin: true,
  target: https://example.com:3000,
});
In case your server is HTTPS and target one is HTTPS as well, you should include SSL certificate

httpProxy.createServer({
  ssl: {
    key: fs.readFileSync('valid-ssl-key.pem', 'utf8'),
    cert: fs.readFileSync('valid-ssl-cert.pem', 'utf8')
  },
  target: 'https://example.com:3000',
  secure: true
}).listen(443);
```

I updated verifier.js (in pact-js that is sitting in node_modules) to include the `changeOrigin` param and my requests are working

```
    // Get the Express app that will run on the HTTP Proxy
    Verifier.prototype.createProxy = function () {
        var _this = this;
        var app = express();
        var proxy = new HttpProxy();
        app.use(this.stateSetupPath, bodyParser.json());
        app.use(this.stateSetupPath, bodyParser.urlencoded({ extended: true }));
        // Allow for request filtering
        if (this.config.requestFilter !== undefined) {
            app.use(this.config.requestFilter);
        }
        // Setup provider state handler
        app.post(this.stateSetupPath, this.createProxyStateHandler());
        // Proxy server will respond to Verifier process
        app.all("/*", function (req, res) {
            logger_1.default.debug("Proxing", req.path);
            proxy.web(req, res, {
                secure: _this.config.validateSSL === true,
                **changeOrigin: true,**
                target: _this.config.providerBaseUrl,
            });
        });
        return app;
    };
```

I am happy to propose a PR (obviously passing it in as opts, rather than hardcoding) , to include this option, as I also wanted to update the readme docs to include the new option `validateSSL` that was included in v8.0.4

I will try and provide a working example, but can't really expose my providers endpoints, and don't have a CA to sign certs in my personal project.

**Comment by mefellows** [comment]
Thanks @YOU54F, I'm happy to bring this in.

**Comment by YOU54F** [comment]
Awesome chap, hope to help out more in the future where I can. 👍

**Comment by mefellows** [comment]
Thanks for your help on this one.

#282 is being released as we speak (`v8.0.5`), closing.
